src/psyclone/tests/dynamo0p3_lma_test.py

```diff
@@ -0,0 +1,869 @@
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford and A. R. Porter, STFC Daresbury Lab
+# Modified I. Kavcic, Met Office
+
+''' This module tests the support for LMA operators in the Dynamo 0.3 API
+using pytest. '''
+
+# imports
+from __future__ import absolute_import, print_function
+import os
+import pytest
+import fparser
+from fparser import api as fpapi
+from psyclone.parse import parse, ParseError
+from psyclone.psyGen import PSyFactory, GenerationError
+from psyclone.dynamo0p3 import DynKernMetadata, DynKern, FunctionSpace
+from psyclone_test_utils import code_compiles, TEST_COMPILE
+
+# constants
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "test_files", "dynamo0p3")
+
+TEST_API = "dynamo0.3"
+
+CODE = '''
+module testkern_qr
+  type, extends(kernel_type) :: testkern_qr_type
+     type(arg_type), meta_args(6) =                 &
+          (/ arg_type(gh_real, gh_read),            &
+             arg_type(gh_field,gh_write,w1),        &
+             arg_type(gh_field,gh_read, w2),        &
+             arg_type(gh_operator,gh_read, w2, w2), &
+             arg_type(gh_field,gh_read, w3),        &
+             arg_type(gh_integer, gh_read)          &
+           /)
+     type(func_type), dimension(3) :: meta_funcs =  &
+          (/ func_type(w1, gh_basis),               &
+             func_type(w2, gh_diff_basis),          &
+             func_type(w3, gh_basis, gh_diff_basis) &
+           /)
+     integer, parameter :: iterates_over = cells
+     integer, parameter :: gh_shape = gh_quadrature_XYoZ
+   contains
+     procedure() :: code => testkern_qr_code
+  end type testkern_qr_type
+contains
+  subroutine testkern_qr_code(a,b,c,d)
+  end subroutine testkern_qr_code
+end module testkern_qr
+'''
+
+
+def test_get_op_wrong_name():
+    ''' Tests that the get_operator_name() utility raises an error
+    if passed the name of something that is not a valid operator '''
+    from psyclone.dynamo0p3 import get_fs_operator_name
+    with pytest.raises(GenerationError) as err:
+        get_fs_operator_name("not_an_op", FunctionSpace("w3", None))
+    assert "Unsupported name 'not_an_op' found" in str(err)
+
+
+def test_ad_op_type_too_few_args():
+    ''' Tests that an error is raised when the operator descriptor
+    metadata has fewer than 4 args. '''
+    fparser.logging.disable(fparser.logging.CRITICAL)
```

Comment:
Is this required anymore? Ditto for other tests in this file.

Reply:
I'm not sure I ever knew what it was required for :-) I've removed it and all the tests still pass so we'll see whether Travis is happy...